src/3d.js

```diff
@@ -326,19 +596,59 @@ canvas:focus { outline: none; }
 				return sprite.room === room.id;
 			});
 			[].concat(items, tiles, sprites).forEach(function (drawing) {
-				requestIdleCallback(function () {
-					var f = drawing.animation.frameIndex;
-					for (var i = 0; i < drawing.animation.frameCount; ++i) {
-						drawing.animation.frameIndex = i;
-						getTexture(drawing, room.pal);
+				// replace drawings marked with the #draw tag
+				if (hackOptions.replaceDrawingTag) {
+					var name = drawing.name || '';
+					var tag = name.match(/#draw[- _]?(TIL|SPR|ITM)[- _]?([a-zA-Z0-9]+)/);
+					if (tag) {
+						// console.log(tag);
+						var map;
+						// tag[1] is the first capturing group, it can be either TIL, SPR, or ITM
+						switch (tag[1]) {
+							case 'TIL':
+								map = bitsy.tile;
+								break;
+							case 'SPR':
+								map = bitsy.sprite;
+								break;
+							case 'ITM':
+								map = bitsy.item;
+								break;
+						}
+						// tag[2] is the second capturing group which returns drawing id
+						var id = tag[2];
+						drawing.drw = map[id].drw;
+						drawing.animation.frameCount = map[id].animation.frameCount;
+						drawing.animation.isAnimated = map[id].animation.isAnimated;
+						drawing.col = map[id].col;
 					}
-					drawing.animation.frameIndex = f;
-				})
+				}
+				// preload textures
+				// safari, edge and ie don't support requestIdleCallback. make sure to check
+				// or it will throw an error and won't run this function on other drawings
+				// and #draw tags won't have any effect
+				if (hackOptions.preloadTextures && window.hasOwnProperty('requestIdleCallback')) {
+					requestIdleCallback(function () {
+						var f = drawing.animation.frameIndex;
+						for (var i = 0; i < drawing.animation.frameCount; ++i) {
+							drawing.animation.frameIndex = i;
+							getTexture(drawing, room.pal);
+						}
+						drawing.animation.frameIndex = f;
+					})
+				}
 			});
 		});
 	}
 });
 
+// to conveniently adjust meshes cause pivot points don't work or they work in some really counterintuitive way
```

Comment:
this is only used during init so not a big concern, but avoiding vertex transformation would be nice since it is more expensive. what's the issues you're having with pivot points? i haven't used them at all in babylon yet but would be good to implement this as intended if we can

Reply:
idk they just don't work.. or at least don't work like pivot points in blender and unity.. and if they are not like this i don't know how they are useful..
like if pivot point is set to the center of the box and i change it to be at the bottom and then set box position form (0,0,0) to (1,0,0) i would expect it to also go 0.5 up.. but it just doesn't. i tried setting pivots and then positioning meshes in all manners of ways (i made template meshes visible to see what's going on) and it looked like they were positioned without taking pivot point into account at all
mb i'm just missing something obvious...
i tried using babylon playground examples but it runs very slow for me and also as far as i remember it didn't work even in the playground

Reply:
from their docs, yeah it sounds like it explicitly does not behave how other 3d software usually does https://doc.babylonjs.com/how_to/pivots:
> The latest pivot in Babylon (v3.2+) is a pre-transformation pivot and behaves differently than in tools like 3DS Max and Maya as the object's position will move if scale is applied prior to setting the pivot. To get pivot behavior that matches these tools it is recommended to set the object as a child of another transform node that will act as the pivot see: https://www.babylonjs-playground.com/#GH4N1R#1

> STRESSED NOTE For those of you who wrote code using a pivot for versions of Babylon.js before v3.2 and who want to update the version of Babylon.js to a currrent one should change each occurance, in their project code, of `mesh.setPivotMatrix(BABYLON.Matrix.Translation(-x, -y, -z));` to `mesh.setPivotMatrix(BABYLON.Matrix.Translation(-x, -y, -z), false);`

i'd suggest taking a quick look at the `setPivotMatrix(..., false)` they recommend to see if it works as expected, but if not prob fine to leave this as-is